### ISSUE TITLE
Target image may need an update

### ISSUE THREAD

**Issue by Ajedi32** [tea-drinker/saveDashie#6]
Looks like there's a few additions to the target image [that are starting to pick up steam](https://www.reddit.com/r/mylittlepony/comments/62qfpd/sleepy_rd_in_rplace_is_done_great_job_everyone/dfpuu1n/).

![](https://i.imgur.com/l62kg16.png)

I'm not really sure how to update the image myself, otherwise I'd send a PR...

**Comment by tea-drinker** [comment]
I'm going to trim the bot to only fix PONIES! and RD herself. Then the background can evolve on its own

**Comment by Ajedi32** [comment]
Seems like a reasonable idea. Though FWIW, there's another project now which is doing something very similar to this: https://github.com/hithroc/fixRD

Rather than rely on the WebSockets connection, it seems like this one just checks every pixel in a random order and replaces any which aren't set correctly. Also automatically updates the target image...

**Comment by tea-drinker** [comment]
Yeah, I saw that one. Working from an image would make things a load easier to maintain and automatic updating is very cool, but I prefer the idea of depending on websockets. The picture is usually mostly correct so fixing the defects keeps it right. Randomly selecting pixels to check will find things my program doesn't know about, but my program should keep things tidy more efficiently.

**Comment by Ajedi32** [comment]
Yeah, websockets does seem like a much more efficient method. Though in practice I'm not actually sure it's quite as reliable.

Anyway, presumably this issue is resolved by 2b635db4f0c5b6817c54439e81a98c93ad6d404a? (And that's why you closed it?)

**Comment by tea-drinker** [comment]
I fixed it in https://github.com/tea-drinker/saveDashie/commit/aa7edf4b27c22b1fdb79d74d6c0a70ae01f5b0e0. Minus 1 is the ignore mask. But I've made a few more updates too. The cutie mark was the subject of a bun fight so I ignored enough to let the competing designs fight it out.

Re: websockets less reliable. Yep. My websocket keeps exploding. However next April fools there will probably be a new sub that depends on a websocket so learning how to handle them is going to be useful. Right now I'm refactoring everything to make it testable and hopefully reusable so I can spin up next year's bot faster.
